client/editor/tool_tile.cpp

```diff
@@ -8,8 +8,11 @@
 #include "editor/tool_tile.h"
 
 // Client
-#include "fcintl.h" //editor_tool_tile
-#include "fonts.h"  //editor_tool_tile
+#include "fcintl.h"         //editor_tool_tile::editor_tool_tile
+#include "fonts.h"          //editor_tool_tile::editor_tool_tile
+#include "mapctrl_common.h" //editor_tool_tile::select_tile
+
+bool is_tile_tool = false;
```

Comment:
Not a fan at all, this will not scale to multiple tools. Imagine you can end up with `is_tile_tool = true` and `is_terrain_tool = true`: what do you do?
Better have a variable in `map_editor` that knows which tool is currently active and calls it as needed.

Reply:
ok that sounds like a good idea. Would also help with the check in mapctrl.

Reply:
I'm going to use the editor tool type enum to cover this

Reply:
got rid of it